lib/menubar/menu_gen.lua

```diff
@@ -83,48 +83,60 @@ local function trim(s)
 end
 
 --- Generate an array of all visible menu entries.
--- @treturn table All menu entries.
-function menu_gen.generate()
+-- @tparam function callback Will be fired when all menu entries will be parsed.
```

Comment:
...when all menu entries *were* parsed.

Also, I wouldn't mind adding `with the resulting list of menu entries as argument`.